### ISSUE TITLE
Recipes defining `outputs` without `package/version` fail in 3.28.0

### ISSUE THREAD

**Issue by kenodegard** [conda/conda-build#5097]
### What happened?

https://github.com/conda/conda-build/pull/5055 corrected how the metadata's version was validated and resulted in multi outputs failing to build unless they also define the top level `package/version`.

Xref https://github.com/conda-forge/admin-requests/pull/890

### Conda Details

<details>
<summary><code>conda info</code></summary>

```
     active environment : None
            shell level : 0
       user config file : /Users/kodegard/.condarc
 populated config files : /Users/kodegard/.condarc
          conda version : 23.11.0
    conda-build version : 3.28.0
         python version : 3.10.13.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=m1
                          __conda=23.11.0=0
                          __osx=13.6=0
                          __unix=0=0
       base environment : /Users/kodegard/.conda/arm64/23.10.0/3.11  (writable)
      conda av data dir : /Users/kodegard/.conda/arm64/23.10.0/3.11/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-arm64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-arm64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/kodegard/.conda/arm64/23.10.0/3.11/pkgs
                          /Users/kodegard/.conda/pkgs
       envs directories : /Users/kodegard/.conda/arm64/23.10.0/3.11/envs
                          /Users/kodegard/.conda/envs
               platform : osx-arm64
             user-agent : conda/23.11.0 requests/2.31.0 CPython/3.10.13 Darwin/22.6.0 OSX/13.6 solver/libmamba conda-libmamba-solver/23.11.1 libmambapy/1.5.3
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

</details>

**Comment by kenodegard** [comment]
Before #5055, where `package/version` (e.g., `None`) is cast to a string (e.g., `"None"`) before validation:

https://github.com/conda/conda-build/blob/923b59aa791e6b97691037d460be479317923104/conda_build/metadata.py#L1449-L1452

After #5055, where `package/version` is validated before cast to a string:

https://github.com/conda/conda-build/blob/decbcf9dd6250dca251e195790b8fb81ec0c5fe9/conda_build/metadata.py#L1443-L1447

**Comment by jezdez** [comment]
Fixed by #5096.
